roles/translator/src/upstream_sv2/upstream.rs

```diff
@@ -375,19 +403,20 @@ impl Upstream {
                     roles_logic_sv2::parsers::Mining::SubmitSharesExtended(sv2_submit),
                 );
 
-                let frame: StdFrame = message
-                    .try_into()
-                    .expect("Err converting `PoolMessage` to `StdFrame`");
+                let frame: StdFrame = handle_result!(tx_status, message.try_into());
                 // Doesnt actually send because of Braiins Pool issue that needs to be fixed
                 info!("Up: Sending: {:?}", &frame);
 
-                let frame: EitherFrame = frame
-                    .try_into()
-                    .expect("Err converting `StdFrame` to `EitherFrame`");
-                let sender = self_
-                    .safe_lock(|self_| self_.connection.sender.clone())
```

Comment:
I'd love to know if there was a good reason to do this every time.

Reply:
not sure what you are referring to. Are you referring to extracting the `channel_id` every loop?

Reply:
I'm referring to the `connection.sender.clone()` - I agree this should be moved outside the loop if it can be, I'm just wondering if there was a reason to put it in the loop originally.